packages/public/common-ui-web/test/components/Slider.test.tsx

```diff
@@ -0,0 +1,164 @@
+  eventHandlers: {
+    onMouseLeave: jest.fn(),
+    onMouseEnter: jest.fn(),
+    onMouseDown: jest.fn(),
+    onMouseUp: jest.fn(),
+  },
+}));
+
+describe('Slider component', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should display the slider without any props', () => {
+    const { unmount } = render(<Slider />);
+
+    expect(screen.getByTestId('slider')).toBeInTheDocument();
+    expect(screen.getByTestId('track')).toBeInTheDocument();
+    expect(screen.getByTestId('value-track')).toBeInTheDocument();
+    expect(screen.getByTestId('thumb')).toBeInTheDocument();
+    expect(screen.getByTestId('hover-thumb')).toBeInTheDocument();
+
+    const sliderEl = screen.getByTestId('slider');
+    expect(sliderEl.style.opacity).toEqual('');
+    expect(sliderEl.style.cursor).toEqual('pointer');
+
+    unmount();
+  });
+
+  it('should not be clickable and have opacity ON when disabled is true', () => {
+    jest.spyOn(common, 'useInteractiveStatus').mockImplementationOnce(() => ({
+      status: InteractiveStatus.DISABLED,
+      eventHandlers: {
+        onMouseLeave: jest.fn(),
+        onMouseEnter: jest.fn(),
+        onMouseDown: jest.fn(),
+        onMouseUp: jest.fn(),
+      },
+    }));
+    const onChange = jest.fn();
+
+    const { unmount } = render(<Slider disabled={true} onChange={onChange} />);
+    const sliderEl = screen.getByTestId('slider');
+    expect(sliderEl.style.opacity).not.toBeNull();
+    expect(sliderEl.style.cursor).toEqual('default');
+
+    fireEvent.mouseDown(sliderEl);
+    fireEvent.click(sliderEl);
+    fireEvent.mouseMove(sliderEl, { clientX: 100 });
+    fireEvent.mouseUp(sliderEl);
+    expect(onChange).not.toHaveBeenCalled();
+
+    unmount();
+  });
+
+  it('should display the thumb in the most right when value is greater than max', () => {
+    const { unmount } = render(<Slider value={200} max={100} />);
+    const thumbEl = screen.getByTestId('thumb');
+    expect(thumbEl.style.left).toEqual('100%');
+
+    unmount();
+  });
+
+  it('should display the thumb in the most left when value is less than min', () => {
+    const { unmount } = render(<Slider value={100} min={200} />);
+    const thumbEl = screen.getByTestId('thumb');
+    expect(thumbEl.style.left).toEqual('0%');
+
+    unmount();
+  });
+
+  it('should call the onChange callback when clicked on and moved on', () => {
+    const onChange = jest.fn();
+
+    const { unmount } = render(<Slider onChange={onChange} />);
+    const sliderEl = screen.getByTestId('slider');
+    fireEvent.mouseDown(sliderEl);
+    fireEvent.click(sliderEl);
+    fireEvent.mouseMove(sliderEl, { clientX: 100 });
+    fireEvent.mouseUp(sliderEl);
+    expect(onChange).toHaveBeenCalledTimes(2);
+
+    unmount();
+  });
+
+  it('should diplay and change left CSSproperty when mouseMove event is trigeer', () => {
+    const onChange = jest.fn();
+    HTMLElement.prototype.getBoundingClientRect = jest.fn(() => ({
```

Comment:
try to replace every `HTMLElement.prototype.getBoundingClientRect` with things like:

```javascript
// Get ref of div
const slider = screen.getByTestId(...);
const spy = jest.spyOn(slider, 'getBoundingClientRect').mockImplementation(() => { ... });

// And then if you need to check if the function was called :
expect(spy).toHaveBeenCalled();
```